internal/services/oracle/oracle_exascale_database_virtual_machine_cluster_resource.go

```diff
@@ -0,0 +1,622 @@
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(1, 255),
+				validate.ExascaleDatabaseVirtualMachineClusterName,
+			),
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"display_name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(1, 255),
+				validate.ExascaleDatabaseVirtualMachineClusterName,
+			),
+		},
+
+		"enabled_ecpu_count": {
+			Type:     pluginsdk.TypeInt,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.IntBetween(8, 200),
+				validation.IntDivisibleBy(4),
+			),
+		},
+
+		"exascale_database_storage_vault_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: exascaledbstoragevaults.ValidateExascaleDbStorageVaultID,
+		},
+
+		"hostname": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"node_count": {
+			Type:         pluginsdk.TypeInt,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IntBetween(2, 10),
+		},
+
+		"shape": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"ssh_public_keys": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+
+		"subnet_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateSubnetID,
+		},
+
+		"total_ecpu_count": {
+			Type:     pluginsdk.TypeInt,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.IntBetween(8, 200),
+				validation.IntDivisibleBy(4),
+			),
+		},
+
+		"virtual_machine_file_system_storage": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"total_size_in_gb": {
+						Type:     pluginsdk.TypeInt,
+						Required: true,
+						ForceNew: true,
+					},
+				},
+			},
+		},
+
+		"virtual_network_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateVirtualNetworkID,
+		},
+
+		"backup_subnet_cidr": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IsCIDR,
+		},
+
+		"cluster_name": {
+			Type: pluginsdk.TypeString,
+			// O+C if not specified, this gets set to the name of the virtual machine
+			Optional: true,
+			Computed: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[a-zA-Z][a-zA-Z0-9-]{0,10}$`),
+				"The Cluster name must begin with an alphabetic character, be no longer than 11 characters, and may contain alphabets, numbers, and hyphens (-).",
+			),
+		},
+
+		"data_collection_option": {
```

Comment:
Based on https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-naming.md, we shall rename it to `data_collection`